build_tools/cmake/iree_native_test.cmake

```diff
@@ -89,6 +92,25 @@ function(iree_native_test)
     list(APPEND _RULE_DATA "${_RULE_TEST_INPUT_FILE_ARG}")
   endif()
 
+  # Detect file location with `{{}}` and handle its portability for all entries
+  # in `_RULE_ARGS`.
+  foreach(_ARG ${_RULE_ARGS})
+    string(REGEX MATCH ".+{{(.+)}}" _FILE_ARG "${_ARG}")
+    if(_FILE_ARG)
+      set(_FILE_PATH ${CMAKE_MATCH_1})
+      list(APPEND _RULE_DATA "${_FILE_PATH}")
+      if (ANDROID)
+        cmake_path(GET _FILE_PATH FILENAME _FILE_BASENAME)
+        set(_FILE_PATH "${_ANDROID_ABS_DIR}/${_FILE_BASENAME}")
+      endif()
+      # remove the `{{}}` from `_ARG` and append it to `_TEST_ARGS`.
+      string(REGEX REPLACE "{{.+}}" "" _FILE_FLAG_PREFIX "${_ARG}")
+      list(APPEND _TEST_ARGS "${_FILE_FLAG_PREFIX}${_FILE_PATH}")
```

Comment:
This assumes the file is at the end of the argument, I think. What about

`string(REGEX REPLACE "{{(.+)}}" "\1" _ARG "${_ARG}")`